src/ledgermonolith/src/main/java/anthos/samples/bankofanthos/ledgermonolith/LedgerMonolithController.java

```diff
@@ -16,36 +16,176 @@
+
+    private JWTVerifier verifier;
+    private LedgerReader ledgerReader;
+
+    // Combined balancereader / txnhistory cache 
+    private LoadingCache<String, AccountInfo> ledgerReaderCache;
+    private Cache<String, Long> ledgerWriterCache; 
 
-    @Value("${VERSION}")
+    private TransactionRepository transactionRepository;
+    private TransactionValidator transactionValidator;
+
+
+    private String localRoutingNum;
     private String version;
 
-   /**
+    public static final String READINESS_CODE = "ok";
+    public static final String UNAUTHORIZED_CODE = "not authorized";
+    public static final String JWT_ACCOUNT_KEY = "acct";
+
+    @Autowired
+    RestTemplate restTemplate;
+
+    /**
+    * Constructor.
+    *
+    * Initializes JWT verifier.
+    * (merges together constructors for LedgerWriter, Transactionhistory, Balancereader. 
+    **ONE** shared DB cache, not multiple)
+    */
+    @Autowired
+    public LedgerMonolithController(
+            @Value("${PUB_KEY_PATH}") final String publicKeyPath,
+            LoadingCache<String, AccountInfo> ledgerReaderCache,
+            JWTVerifier verifier,
+            TransactionRepository transactionRepository, 
+            TransactionValidator transactionValidator,  
+            LedgerReader reader, 
+            @Value("${LOCAL_ROUTING_NUM}") String localRoutingNum,
+            @Value("${VERSION}") String version) {
+        this.verifier = verifier;
+        this.transactionRepository = transactionRepository;
+        this.transactionValidator = transactionValidator;
+        this.localRoutingNum = localRoutingNum;
+        this.version = version;
+
+        // balance reader 
+        this.ledgerReaderCache = ledgerReaderCache;
+
+        // ledger writer  
+        this.ledgerWriterCache = CacheBuilder.newBuilder()
+        .recordStats()
+        .expireAfterWrite(1, TimeUnit.HOURS)
+        .build();
+
+        // Ledger Cache processing  
+        this.ledgerReader = reader;
+        this.ledgerReader.startWithCallback(transaction -> {
+            final String fromId = transaction.getFromAccountNum();
+            final String fromRouting = transaction.getFromRoutingNum();
+            final String toId = transaction.getToAccountNum();
+            final String toRouting = transaction.getToRoutingNum();
+            final Integer amount = transaction.getAmount();
+
+            if (fromRouting.equals(localRoutingNum)
+                && this.ledgerReaderCache.asMap().containsKey(fromId)) {
+                AccountInfo info = ledgerReaderCache.asMap().get(fromId); 
+                Long prevBalance = info.getBalance(); 
+                Long newBalance = prevBalance - amount; 
+                processTransaction(fromId, newBalance, transaction);
+            }
+
+            if (toRouting.equals(localRoutingNum)
+                && this.ledgerReaderCache.asMap().containsKey(toId)) {
+                AccountInfo info = ledgerReaderCache.asMap().get(fromId); 
```

Comment:
@djmailhot  I found the bug in my implementation, it was getting the wrong value from the cache. This should have been `toId`, not `fromId`. :)